api/client/hijack.go

```diff
@@ -196,19 +193,15 @@ func (cli *DockerCli) hijack(method, path string, setRawTerminal bool, in io.Rea
 		defer term.RestoreTerminal(cli.inFd, oldState)
 	}
 
+	receiveStdout := make(chan error)
```

Comment:
Yes. BUT.. I think @noxiouz is right, because channel with value is cheaper than whole closure. With buffered channel you can just return from closure and forget about it.